src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -68,13 +69,19 @@ private static Kryo createKryo() {
 	@Override
 	public void save( final URI identifier, final T element ) throws IOException {
 		final Path path = Paths.get( identifier );
+		final File file = path.toFile();
 		FileUtil.ensureFolder( path.toFile() );
 		try ( final Output output = new Output( Files.newOutputStream( path ) ) ) {
 			output.writeString( version );
 			logger.debug( "Writing {} to {}. Do not write to same identifier or interrupt until done.", element,
 					identifier );
 			kryo.writeClassAndObject( output, element );
 			logger.debug( "Done writing {} to {}", element, identifier );
+		} catch ( Error | Exception anything ) {
+			logger.warn( "Error writing to file {}. Delete what has been writen to not leave corrupt file behind...",
+					identifier, anything );
+			FileUtil.deleteSecurely( file );
```

Comment:
I suggest to remove `deleteSecurely` and reuse Apache's [`deleteQuietly`](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)), which we already use in `FilteUtil`:

>Deletes a file, never throwing an exception. If file is a directory, delete it and all sub-directories.

Reply:
Done